crates/re_space_view_spatial/src/space_view_2d.rs

```diff
@@ -316,121 +278,123 @@ fn count_non_nested_entities_with_component(
             .children
             .values()
             .map(|child| {
-                count_non_nested_entities_with_component(entity_bucket, child, component_name)
+                count_non_nested_entities_with_component(
+                    image_dimensions,
+                    entity_bucket,
+                    child,
+                    component_name,
+                )
             })
             .sum()
     }
 }
 
-/// Given a bucket of image entities.
-fn add_recommended_space_views_for_image_bucket(
-    ctx: &ViewerContext<'_>,
+// Find the image dimensions of every image-entity in the bucket that is not
+// not nested under another image.
+//
+// We track a set of just height/width as different channels could be allowed to
+// stack.
+fn find_non_nested_image_dimensions(
+    image_dimensions: &IntMap<EntityPath, ImageDimensions>,
     entity_bucket: &IntSet<EntityPath>,
+    subtree: &EntityTree,
+    found_image_dimensions: &mut HashSet<(u64, u64)>,
```

Comment:
nitpick: I usually find `[u64; 2]` more natural for things like these, since the two types here are the same by necessity, while tuples for for things where the types are different (at least "logically")

```suggestion
    found_image_dimensions: &mut HashSet<[u64; 2]>,
```